### DISCUSSION TITLE
Exception / Device

### DISCUSSION THREAD

**Discussion by xstrex** [elchininet/custom-sidebar#248] [Q&A]
Trying to make a device specific sidebar, and it looks like I can accomplish this with an `exception:` as documented [here](https://github.com/elchininet/custom-sidebar#exceptions). Yet the description says it's using a device name, or the user-agent string. 

I'm using the HA companion app on iOS, and don't know what the actual user-agent would be. I've tried `-iphone` and `- com.homeassistant.companion` neither of which has worked. Also searched the forum, and the companion [docs](https://companion.home-assistant.io/docs/getting_started/) neither of which has yielding any results. Here's another related, unanswered forum [post](https://community.home-assistant.io/t/ios-companion-app-user-agent/764078). 

I'm not sure how to pull the user-agent from the app, or HA, since the app isn't exactly a web-browser, and doesn't have a developer console (like a web browser would). Any suggestions? How would I make a device exception for the companion app?

**Comment by elchininet** [comment]
Hi @xstrex,
The device's name uses the browser userAgent. The App is basically a web-browser but embedded in [a WebView](https://developer.apple.com/documentation/webkit/wkwebview). The WebView, as it uses a web-browser engine should have also [its own userAgent](https://gist.github.com/taylorhughes/dae564f63b3d702429a5f37bd22b97b8), and most probably it contains the `iphone` string if it is an iPhone.

First of all, before trying to reach to Home Assistant iOS developers to get what is the possible user agent of the App, let's try to check if `custom-sidebar` is really not being applied using `iphone` as `device`. Create a minimum configuration example without exceptions (something like a configuration adding a single item), [clean the cache of your App](https://github.com/NemesisRE/kiosk-mode/issues/112#issuecomment-1739876659), close and open the App and and check if the new item is added correctly to the sidebar. If it does, create an exception adding another item, clean cache again, close and open the App, and check if the new exception item is added.

Let me know your results.

**Comment by TotalPanther317** [comment]
I wanted to do the same but on android previously but did not find how to anywhere. What is the android app user agent? And how do I enter it in the configuration?

**Reply by elchininet** [reply]
Hi @TotalPanther317,
I would start with "android" :)

https://stackoverflow.com/a/13572799/2035354

**Reply by elchininet** [reply]
About cleaning the Webview cache in Android it is not as straightforward as in an iOS device. 

https://github.com/elchininet/custom-sidebar/issues/130#issuecomment-2327025323 

The "easiest" is to log-out and log-in again but that would be problematic with sensors, device ids, etc.

**Reply by TotalPanther317** [reply]
'android' seems to be working but I am not sure yet.

**Reply by elchininet** [reply]
Just make a simple tests like the one that I described before. A single item in an exception is a good test.

**Reply by TotalPanther317** [reply]
I have checked the app again now and it works! "android" is the correct user agent.
Thank you.

**Reply by elchininet** [reply]
> I have checked the app again now and it works! "android" is the correct user agent.
> Thank you.

Glad to hear that it is working 👍
